### PR TITLE
added horizontal_obstructed

### PR DESCRIPTION
**Pull request by varghes1**
horizontally_obstructed? logic written. tests written and passing. Same logic as vertical obstruction. If someone can okay it I can close PR.

### CODE REVIEW

File: spec/models/chess_piece_spec.rb [hunk 1]
```diff
@@ -53,6 +53,22 @@
...
+      let(:game) {Game.create}
+      let(:piece) {ChessPiece.create(position_x: 4, position_y: 4, game: game)}
+      let(:second_piece) {ChessPiece.create(position_x: 6, position_y: 4, game: game)}
+      it "will return true when trying to move to [7,4]" do
+        second_piece
+        expect(piece.is_obstructed?(7,4)).to be_truthy
+      end
+      it "will return false when trying to move to [5,4]" do
+        second_piece
+        expect(piece.is_obstructed?(5,4)).to be_falsey
```
Comment (KaoruDev) [inline spec/models/chess_piece_spec.rb:67]: forgot to ask you for `vertical` but uhh what happens when you try `(2, 4)`? Can you write a spec for it?

**Comment by varghes1** [comment]
vertical test added and passing.

File: spec/models/chess_piece_spec.rb [hunk 2]
```diff
@@ -53,6 +53,38 @@
...
+      let(:game) {Game.create}
+      let(:piece) {ChessPiece.create(position_x: 4, position_y: 4, game: game)}
+      let(:second_piece) {ChessPiece.create(position_x: 6, position_y: 4, game: game)}
+      it "will return true when trying to move to [7,4]" do
+        second_piece
+        expect(piece.is_obstructed?(7,4)).to be_truthy
+      end
+      it "will return false when trying to move to [2,4]" do
+        second_piece
+        expect(piece.is_obstructed?(2,4)).to be_falsey
```
Comment (KaoruDev) [inline spec/models/chess_piece_spec.rb]: what if there is a piece at `(3,4)`? it should return `true`

i.e.

```
it "will return true when trying to move to [2,4] if there is a pieces on (3, 4)" do
  ChessPiece.create(position_x: 3, position_y: 4, game: game)
  expect(piece.is_obstructed?(2,4)).to be_truthy
end
```

**Comment by varghes1** [comment]
it will run properly if the pieces want to go front/back, up/down for the horizontal and vertical moves.

**Comment by jmundraw** [comment]
looks good to merge to me

**Pull request merged**